File: src/sse.js
import { assertValidSessionEvent } from "./session-event.js";

export function formatSseEvent(event) {
  assertValidSessionEvent(event);
  return [
    `id: ${event.eventId}`,
    `event: ${event.type}`,
    ...formatDataLines(JSON.stringify(event)),
    ""
  ].join("\n") + "\n";
}

export function formatSseRetry(milliseconds) {
  if (!Number.isSafeInteger(milliseconds) || milliseconds < 0) {
    throw new TypeError("milliseconds must be a non-negative safe integer");
  }
  return `retry: ${milliseconds}\n\n`;
}

function formatDataLines(serializedEvent) {
  return serializedEvent.split(/\r?\n/u).map((line) => `data: ${line}`);
}

File: src/ai_assist_session_events/sse.py
from __future__ import annotations

import json

from .session_event import assert_valid_session_event


def format_sse_event(event: dict) -> str:
    assert_valid_session_event(event)
    serialized_event = json.dumps(event, separators=(",", ":"))
    lines = [
        f"id: {event['eventId']}",
        f"event: {event['type']}",
        *[f"data: {line}" for line in serialized_event.splitlines()],
        "",
    ]
    return "\n".join(lines) + "\n"


def format_sse_retry(milliseconds: int) -> str:
    if not isinstance(milliseconds, int) or isinstance(milliseconds, bool) or milliseconds < 0:
        raise TypeError("milliseconds must be a non-negative safe integer")
    return f"retry: {milliseconds}\n\n"
